bin/describe.ml

```diff
@@ -28,38 +28,159 @@ let man =
 
 let info = Term.info "describe" ~doc ~man
 
-(* Crawl the workspace to get all the data *)
+(** The module [Descr] is a typed representation of the description of a
+    workspace, that is provided by the ``dune describe workspace`` command.
+
+    Each sub-module contains a [to_dyn] function, that translates the
+    descriptors to a value of type [Dyn.t].
+
+    The typed representation aims at precisely describing the structure of the
+    information computed by ``dune describe``, and hopefully make users' life
+    easier in decoding the S-expressions into meaningful contents. *)
+module Descr = struct
+  (** [dyn_path p] converts a path to a value of type [Dyn.t] *)
+  let dyn_path (p : Path.t) : Dyn.t = Dyn.String (Path.to_string p)
```

Comment:
I believe this is `Path.to_dyn p`?

Reply:
That's correct. This piece of code was present already before my patch, so I have kept it as it was.
I will push a fix to use ``Path.to_dyn`` instead.

Reply:
Fixed.

Reply:
Actually, I reverted this fix, because the code above is not equivalent to ``Path.to_dyn``. The latter would produce something like ``(In_build_dir "the_path")``.